src/modules/auth/reducers.js

```diff
@@ -0,0 +1,12 @@
+import createReducer from '../../utils/create-reducer'
+
+const INITIAL_STATE = null
+
+
+export const token = createReducer(INITIAL_STATE, {
+  AUTHENTICATE_WITH_GOOGLE: (state, action) => {
+    const { payload, error } = action
+    if (error) return state
```

Comment:
That pattern also excludes the need for check for error and return state if any.